src/libs/actions/Policy.js

```diff
@@ -319,12 +366,28 @@ function invite(logins, welcomeNote, policyID) {
     const newEmployeeList = _.map(logins, login => OptionsListUtils.addSMSDomainIfPhoneNumber(login));
 
     // Make a shallow copy to preserve original data, and concat the login
-    const policy = _.clone(allPolicies[key]);
-    policy.employeeList = [...policy.employeeList, ...newEmployeeList];
-    policy.alertMessage = '';
+    const policy = allPolicies[key];
+
+    // Always insert invited logins to pendingInvitations first
+    const pendingInvitations = _.uniq([...policy.pendingInvitations, ...newEmployeeList]);
+    const alertMessage = '';
 
     // Optimistically add the user to the policy
-    Onyx.merge(key, policy);
+    Onyx.merge(key, {pendingInvitations, alertMessage});
+
+    const revertPendingInvitationsList = () => {
+        const policyDataWithoutPending = _.clone(allPolicies[key]);
+        policyDataWithoutPending.pendingInvitations = _.without(
+            policyDataWithoutPending.pendingInvitations, ...newEmployeeList,
+        );
+
+        Onyx.set(key, policyDataWithoutPending);
+    };
+
+    const showGrowlError = (additionalMessage = '') => {
+        const errorMessage = `${Localize.translateLocal('workspace.invite.genericFailureMessage')} ${additionalMessage}`;
```

Comment:
This we will have to confirm first. We are prefixing generic error before any error.